File: packages/smaccmpilot/entry-fn.py
"""
Copyright (C) 2014, NICTA, Inc.
All Rights Reserved.

This software is released under the "BSD3" license.  Read the file
"LICENSE" for more information.
"""

from prj import Module


class EntryFnModule(Module):
    xml_schema = """<schema></schema>"""

    files = [
        {'input': 'entry-fn.h', 'render': True},
        {'input': 'entry-fn.c', 'render': True, 'type': 'c'},
    ]

module = EntryFnModule()

File: packages/smaccmpilot/entry-fn.c
/*
 * Copyright (C) 2014, NICTA, Inc.
 * All Rights Reserved.
 *
 * This software is released under the "BSD3" license.  Read the file
 * "LICENSE" for more information.
 */

#include <rtos-kochab.h>
#include "entry-fn.h"

{{#rtos.tasks}}
extern void {{function}}(void);
{{/rtos.tasks}}

void *entry_fn[{{rtos.tasks.length}}] = {
{{#rtos.tasks}}
    [{{rtos.prefix_const}}TASK_ID_{{name|u}}] = (void*)&{{function}},
{{/rtos.tasks}}
};
